redis/3-highly-available/usr/local/share/redis_functions.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+function get_existing_master()
+{
+  redis-cli -h redis-sentinel -p 26379 --csv SENTINEL get-master-addr-by-name mymaster | tr ',' ' ' | cut -d' ' -f1 | cut -d'"' -f2
+}
+
+function sentinels()
+{
+  # Print out the info of other sentinels
+  redis-cli -h redis-sentinel -p 26379 --csv SENTINEL sentinels mymaster
+}
+
+function master_failover()
+{
+  redis-cli -h redis-sentinel -p 26379 SENTINEL failover mymaster
+}
+
+function wait_for_exit()
+{
+  local WAIT_FOR="$1"
+  wait_for_remote_ports_to_close 120 "$WAIT_FOR"
+}
+
+function sentinel_cleanup()
+{
+  local EXISTING_MASTER="$1"
+  # Run reset on each sentinel.
+  SENTINELS="$(echo -e "SENTINEL sentinels mymaster\nSENTINEL RESET mymaster\n" | redis-cli -h redis-sentinel -p 26379 --csv | sed 's/"name"/\n"name"/g' | grep "name" | grep -v 's-down-time' | sed -E 's/.*"ip","([^"]+)".*/\1/')"
+  if [ -n "$EXISTING_MASTER" ]; then
+    SENTINELS="$(echo "$SENTINELS" | grep -v "$EXISTING_MASTER")"
+  fi
+  for sentinel in $SENTINELS; do
+    sleep 11s
```

Comment:
where does that come from? :-)

Reply:
Added some explanation - though perhaps we can make it more objective than time-based, e.g. query if the previously reset sentinel has some other sentinel knowledge now. For another day perhaps.